### DISCUSSION TITLE
AppImage / snap / docker image?

### DISCUSSION THREAD

**Discussion by ndahn** [inviwo/inviwo#1354] [General]
It would be awesome if you could provide an AppImage / snap / docker / whatever is in fashion right now! I don't see a good reason for having to compile the entire application on top of having to deal with Qt and other dependencies...

**Comment by ResearchDaniel** [comment]
I think that this docker image might work:
https://github.com/inviwo/inviwo/tree/master/tools/docker

Perhaps it could be made more visible

**Comment by ndahn** [comment]
Ah, so there is something already... Even though the instructions on the page are a bit cryptic, that's great to hear! I wasn't aware of that at all. Thank you, Daniel!

Unfortunately, it didn't work for me either :/ I got the following output from `docker build .`:

```
[...]
Setting up libboost-all-dev (1.71.0.0ubuntu2) ...
Processing triggers for libc-bin (2.31-0ubuntu9.9) ...
Removing intermediate container 84610623ad64
 ---> b3229a1cff14
Step 11/26 : COPY . ${IVW_SOURCE_DIR}
 ---> 3d55bef74ba6
Step 12/26 : WORKDIR ${IVW_BUILD_DIR}
 ---> Running in 0433df82ef19
Removing intermediate container 0433df82ef19
 ---> 137a3d589990
Step 13/26 : RUN cmake -DCMAKE_INSTALL_PREFIX:PATH=${IVW_INSTALL_DIR} ${IVW_SOURCE_DIR}
 ---> Running in f865444f0b5e
CMake Error: The source directory "/usr/src/inviwo" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
The command '/bin/sh -c cmake -DCMAKE_INSTALL_PREFIX:PATH=${IVW_INSTALL_DIR} ${IVW_SOURCE_DIR}' returned a non-zero code: 1
```

**Comment by ResearchDaniel** [comment]
Yes, it is a bit cryptic and I have not used it myself so I unfortunately cannot elaborate.

@xeTaiz Do you have an alternative docker image up and running?

**Reply by xeTaiz** [reply]
> Yes, it is a bit cryptic and I have not used it myself so I unfortunately cannot elaborate.
> 
> @xeTaiz Do you have an alternative docker image up and running?

Sorry for the late reply. No I have no docker image running, also I think it would be difficult to get the GUI exposed from within the container. 
I think easy Linux installation wise flatpak is the fancy way to do it these days, but I have no experience in creating those

**Comment by petersteneteg** [comment]
For windows there are also binary installers https://github.com/inviwo/inviwo/releases/tag/latest

**Comment by ndahn** [comment]
I saw those, thank you :) But constantly switching between your data generation on linux and visualization on windows is not a great option... 

I understand if its not a priority for you right now, but at least for me the accessibility makes me look for alternatives and I imagine others would feel the same. From what I saw I much prefer Inviwo over Voreen, so I would really like to use it! Unfortunately I don't have the time to fix dependency issues for days :/

**Comment by petersteneteg** [comment]
@ResearchDaniel I think we should try and create an AppImage, I think that should work quite well, and it seems to support what we need. https://docs.appimage.org/index.html
there seems to be supports for embedding an isolated python dist into the image as well which would be very useful https://github.com/linuxdeploy/awesome-linuxdeploy#linuxdeploy-plugins

**Reply by ResearchDaniel** [reply]
That would be nice. If I recall correctly, it works similar to MacOS app creation (macdeployqt). Since we revamped the support for macdeployqt I think it might be easier to get AppImage to work as well.

**Reply by ndahn** [reply]
I tried that yesterday (installed Qt into the ext/qt folder) using the [AppImage from source](https://docs.appimage.org/packaging-guide/from-source/native-binaries.html#packaging-from-source) method. Inviwo compiled, but during the `make install` step the fmt headers could not be found and I was not able to symlink the folders in a way that would make it work. Looking forward to what you come up with!
